### DISCUSSION TITLE
Avoid certain parcels being on the same route

### DISCUSSION THREAD

**Discussion by robladbrook** [google/or-tools#3737] [Routing (and legacy CP) questions] [answered]
Is there a simple way of avoiding certain parcels being on the same route?

Let's say I have 500 parcels I'd like to deliver using X vehicles.

450 of the parcels can go on any route / any vehicle - let's call these the GENERIC type parcels.
10 parcels are TYPE_A and can only be combined with other TYPE_A parcels or GENERIC type parcels.
10 parcels are TYPE_B and can only be combined with other TYPE_B parcels or GENERIC type parcels.
10 parcels are TYPE_C and can only be combined with other TYPE_C parcels or GENERIC type parcels.
10 parcels are TYPE_D and can only be combined with other TYPE_D parcels or GENERIC type parcels.
10 parcels are TYPE_E and can only be combined with other TYPE_E parcels or GENERIC type parcels. 

The different types can be split across multiple routes, they just can't be mixed together.

How can I enforce this constraint? Thank you!

I'm exploring something along the lines of this (pseudo code) but it doesn't feel right / possible:
```py
for i in type_a_indexes:
    solver().Add(routing.VehicleVar(i) not in [routing.VehicleVar(o) for o in type_b_c_d_e_indexes])
for i in type_b_indexes:
    solver().Add(routing.VehicleVar(i) not in [routing.VehicleVar(o) for o in type_a_c_d_e_indexes])
# etc
```

**Comment by Mizux** [comment, marked as answer]
I would:
Create a booleanVar for each vehicle `i`, for each type `k`: `vi_k` which is true if vehicle `i` visit any node of type `k`, etc....
Add a constraint that for each vehicle `i` you have  `sum(vi_a, vi_b, vi_c, vi_d, vi_e) <= 1`

**Reply by robladbrook** [reply]
Thank you Mizux!
